lnwallet/chainfee/estimator.go

```diff
@@ -792,6 +793,12 @@ func NewWebAPIEstimator(api WebAPIFeeSource, noCache bool,
 func (w *WebAPIEstimator) EstimateFeePerKW(numBlocks uint32) (
 	SatPerKWeight, error) {
 
+	// If the estimator hasn't been started yet, we'll return an error as
+	// we can't provide a fee estimate.
+	if !w.started.Load() {
```

Comment:
Or we could move away from start methods entirely and just start it in the constructor when it is called. IMO we should only have `Start` methods for things that primarily interact through channels.

Reply:
RAII can be nice when using OOP. For instance in python you can define what happens when initializing the object in its `__ini__`. Here we may achieve something similar using the `NewXXX` method, which is better than embedding the `Start` method in the constructor (IIUR about this term) as, there might be multiple constructors, we need to carry this concern in every one of them, just like @ellemouton pointed out that the `started` now has to be checked on every API.

Also this pattern has already been used in some of the RPC calls, so we make sure the server is fully started before responding to requests. This led me to think about what `NewXXX` and `Start` should accomplish, maybe we should `start` inside the `NewXXX` so we can be somewhat certain that a given state is reached once the struct is returned. However that's a bigger refactor so I kinda let it go, maybe worth revisiting when it becomes an issue someday (for instance #8166)

Reply:
IMO the `Start`/`Stop` APIs only make sense if we can create a state machine that is capable of multi-start/multi-stop. However, we have (for some reason) intentionally avoided this. Absent multi-start/multi-stop, the lifetime of the object should dictate the lifetime of the goroutines it launches. Therefore one of the things we can do here is launch the goroutines in whatever the `NewXXX` method is, and then hook into the GC to call `XXX.stop()` when all of the `XXX` refs go out of scope using a [gc finalizer](https://pkg.go.dev/runtime#SetFinalizer)

Definitely agree that this is not the place for that much larger discussion to unfold though. Just wanted to share my thoughts as I have found that the `Start`/`Stop` pattern in the codebase to be a wart that doesn't make sense, and people seem to just follow the pattern because "that's what we've always done".